pytr/transactions.py

```diff
@@ -57,3 +61,42 @@ def export_transactions(input_path, output_path, lang="auto", sort=False, date_i
         f.write(lines)
 
     log.info("Deposit creation finished!")
+
+
+class TransactionsJsonExporter:
+    def __init__(self, tr, output_path, not_before):
+        self.tr = tr
+        self.output_path = output_path
+        self.not_before = not_before
+        self.transactions = []
+
+    async def loop(self):
+        await self.tr.timeline_transactions()
+        while True:
+            _subscription_id, subscription, response = await self.tr.recv()
+
+            if subscription["type"] == "timelineTransactions":
+                self.transactions.extend(response["items"])
+
+                # Transactions in the response are ordered from newest to oldest
+                # If the oldest (= last) transaction is older than what we want, exit the loop
+                t = self.transactions[-1]
+                if datetime.fromisoformat(t["timestamp"]) < self.not_before:
+                    return
+
```

Comment:
I've tested this locally, and if you go far back enough to include the first transaction, it just keeps going over the timeline forever because there is no "after" after the chunk with the first transaction is returned.

```suggestion
                if not response["cursors"]["after"]:
                    break

```

Reply:
Committed this change to unblock request in #173